src/main/java/com/fauna/codec/codecs/StreamRequestCodec.java

```diff
@@ -0,0 +1,45 @@
+package com.fauna.codec.codecs;
+
+import com.fauna.codec.CodecProvider;
+import com.fauna.codec.UTF8FaunaGenerator;
+import com.fauna.codec.UTF8FaunaParser;
+import com.fauna.exception.ClientException;
+import com.fauna.stream.StreamRequest;
+
+import java.io.IOException;
+
+public class StreamRequestCodec extends BaseCodec<StreamRequest> {
```

Comment:
I avoided doing this for query, mainly because I didn't want devs or users to be able to encode these types successfully in any form of nesting–it felt like it established a clear line that the requests are top level only.

Let's discuss if you don't agree–I want to make sure we follow a common pattern at least.

Reply:
Yeah, I agree on following a pattern. If I'm understanding correctly, you're saying that `StreamRequestCodec` should not exist, I should just do those encode steps in `buildStreamRequest`.